src/expr/src/expr/expr_regexp.rs

```diff
@@ -248,3 +251,377 @@ impl Expression for RegexpMatchExpression {
+                        _ => {
+                            return Err(ExprError::UnsupportedFunction(
+                                "non-constant in regexp_replace".to_string(),
+                            ))
+                        }
+                    };
+
+                    if n_flag {
+                        // Check if any flag is specified
+                        if let Some(flag_node) = children.next() {
+                            // Get the flag
+                            flags = match &flag_node.get_rex_node()? {
+                                RexNode::Constant(flag_value) => {
+                                    let flag_datum = deserialize_datum(
+                                        flag_value.get_body().as_slice(),
+                                        &DataType::from(flag_node.get_return_type().unwrap()),
+                                    )
+                                    .map_err(|e| ExprError::Internal(e.into()))?;
+
+                                    match flag_datum {
+                                        Some(ScalarImpl::Utf8(v)) => v.to_string(),
+                                        // NULL replacement
+                                        // FIXME: Do we need the NULL match arm here?
+                                        _ => bail!("Expected flag to be a String"),
+                                    }
+                                }
+                                _ => {
+                                    return Err(ExprError::UnsupportedFunction(
+                                        "non-constant in regexp_replace".to_string(),
+                                    ))
+                                }
+                            };
+                        }
+                    } else {
+                        flags = placeholder;
+                    }
+                }
+            }
+        }
+
+        // TODO: Any other error handling?
+        if let Some(_other) = children.next() {
+            // There should not any other option after the `flags`
+            bail!("invalid parameters specified in regexp_replace");
+        }
+
+        // Construct the final `RegexpReplaceExpression`
+        let ctx = RegexpContext::new(&pattern, &flags)?;
+
+        let mut global_flag = false;
+        if !flags.is_empty() && flags.starts_with('g') {
+            // Set the `global_flag` to true
+            global_flag = true;
+        }
+
+        Ok(Self {
+            source,
+            ctx,
+            replacement,
+            return_type,
+            start,
+            n,
+            global_flag,
+        })
+    }
+}
+
+impl RegexpReplaceExpression {
+    /// Match and replace one row, return the replaced string
+    fn match_row(&self, text: Option<&str>) -> Option<String> {
+        if let Some(text) = text {
+            // The start position to begin the search
+            let start = if self.start != -1 { self.start - 1 } else { 0 };
```

Comment:
panicked in line 516 if `self.start <= -2`.

Please test the behavior in pg for such case.

I guess the behavior is similar to https://doc.rust-lang.org/std/primitive.usize.html#method.saturating_sub, but you should verify it.